### PR TITLE
repeater: add reset cmd and modify other cmds

### PR DESCRIPTION
**Pull request by zhang-3**
add reset cmd - use pin INT to reset
    modify button cmd
    modify led cmd - add led3
    
    Signed-off-by: hazel.zhang <hazel.zhang@unisoc.com>

### CODE REVIEW

File: repeater/src/test.c [hunk 1]
```diff
@@ -132,8 +133,17 @@ static int cmd_led(const struct shell *shell, size_t argc, char **argv)
...
+	led = atoi(argv[1]);
 	on_off = atoi(argv[2]);
+	led_pin = CONFIG_LED_PIN1;
+	switch (led) {
+	case 1:
+		led_pin = CONFIG_LED_PIN1; break;
+	case 2:
+		led_pin = CONFIG_LED_PIN2; break;
+	case 3:
+		led_pin = CONFIG_LED_PIN3; break;
```
Comment (easonxiang) [inline repeater/src/test.c]: Add default case to check input parameter.

File: repeater/src/test.c [hunk 2]
```diff
@@ -164,22 +174,52 @@ static void led3_on(struct device *port, struct gpio_callback *cb,
...
 
+#define FUNC3 3
+#define PIN10 10
+#define PULLUP 1
+void reset_misc(void)
+{
+	struct device *pm_dev, *p0_dev;
+
+	pm_dev = device_get_binding(PMUX_DEV);
+	p0_dev = device_get_binding(GPIO_PORT0);
```
Comment (easonxiang) [inline repeater/src/test.c]: Need to check if dev is valid.

File: repeater/src/test.c [hunk 3]
```diff
@@ -189,8 +229,10 @@ SHELL_CREATE_STATIC_SUBCMD_SET(sub_test)
 			"Usage: sleep <ms>", cmd_sleep),
 	SHELL_CMD(led, NULL, "Control led to turn on/off.",
```
Comment (easonxiang) [inline repeater/src/test.c]: Can you add some command usage description, e.g. Usage: led <num> <on/off>

File: repeater/src/test.c [hunk 4]
```diff
@@ -164,22 +174,52 @@ static void led3_on(struct device *port, struct gpio_callback *cb,
 
 static int cmd_button(const struct shell *shell, size_t argc, char **argv)
 {
-	struct device *gpio;
+	struct device *gpio, *pmx_dev;
 
 	gpio = device_get_binding(GPIO_PORT0);
+	pmx_dev = device_get_binding(PMUX_DEV);
 	count = 0;
 	gpio_pin_configure(gpio, BUTTON_PIN, (GPIO_DIR_IN | GPIO_INT
```
Comment (bublover) [inline repeater/src/test.c]: Check if pointer is proper.

File: repeater/src/test.c (hunk 3 above)
Comment (bublover) [inline repeater/src/test.c]: Can you add some command usage description, e.g. Usage: led <pin> <on/off>

File: repeater/src/test.c [hunk 5]
```diff
@@ -164,22 +174,52 @@ static void led3_on(struct device *port, struct gpio_callback *cb,
 
 static int cmd_button(const struct shell *shell, size_t argc, char **argv)
 {
-	struct device *gpio;
+	struct device *gpio, *pmx_dev;
 
 	gpio = device_get_binding(GPIO_PORT0);
+	pmx_dev = device_get_binding(PMUX_DEV);
```
Comment (bublover) [inline repeater/src/test.c]: Check if pointer is proper.

File: repeater/src/test.c [hunk 6]
```diff
@@ -189,8 +229,10 @@ SHELL_CREATE_STATIC_SUBCMD_SET(sub_test)
 			"Usage: sleep <ms>", cmd_sleep),
 	SHELL_CMD(led, NULL, "Control led to turn on/off.",
 			cmd_led),
-	SHELL_CMD(button, NULL, "Push button to control led3",
+	SHELL_CMD(button, NULL, "Push button to control led2",
```
Comment (bublover) [inline repeater/src/test.c]: Can you add some command usage description, e.g. Usage: button <...>

**Comment by zhang-3** [comment]
invalid

**Pull request closed without merging**